kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,895 @@
+     * @details This function is responsible for instantiating a linear solver object using the configuration specified in the `Settings` parameter. It leverages a factory pattern to select and construct the appropriate solver type based on the parameters provided. The selection process considers the solver type, tolerance, maximum iterations, and other solver-specific parameters contained within `Settings`.
+     * The factory supports constructing various types of solvers, such as Conjugate Gradient, LU Decomposition, and others, depending on the `solver_type` parameter within `Settings`. It throws a `std::invalid_argument` exception if the settings specify an unsupported solver type or if required parameters for the selected solver type are missing or invalid.
+     * @param Settings A `Parameters` object containing the configuration settings for the linear solver to be constructed. This includes type, tolerance, maximum iterations, and other solver-specific parameters.
+     * @return Returns a `LinearSolverPointer` pointing to the newly constructed linear solver instance. If the factory implementation cannot find a matching solver type or if essential parameters are missing or invalid, it throws an exception.
+     * @exception std::invalid_argument Thrown if the settings specify an unsupported solver type or if required parameters are missing or invalid.
+     */
+    LinearSolverPointer ConstructLinearSolverFromSettings(const Parameters Settings)
+    {
+        // Implementation that delegates to a factory method.
+        // The factory method 'Create' is responsible for parsing 'Settings'
+        // and constructing the appropriate solver.
+        const auto linear_solver_factory = LinearSolverFactoryType();
+        Parameters linear_solver_settings(Settings);
+        const std::string linear_solver_type = linear_solver_settings["solver_type"].GetString();
+        // If the solver type is a faster direct solver, try to use it
+        if (linear_solver_type == "faster_direct_solver") {
+            std::vector<std::string> faster_direct_solvers;
+            if constexpr (TSparseSpaceType::IsDistributed()) {
+                faster_direct_solvers = std::vector<std::string>({ // May need to be updated and reordered. In fact I think it depends of the size of the equation system
+                    "mumps2",         // Amesos2 (if compiled with MUMPS-support)
+                    "mumps",          // Amesos (if compiled with MUMPS-support)
+                    "super_lu_dist2", // Amesos2 SuperLUDist (if compiled with MPI-support)
+                    "super_lu_dist",  // Amesos SuperLUDist (if compiled with MPI-support)
+                    "amesos2",        // Amesos2
+                    "amesos",         // Amesos
+                    "klu2",           // Amesos2 KLU
+                    "klu",            // Amesos KLU
+                    "basker"          // Amesos2 Basker
```

Comment:
this should be in the trilinos app, do you think it could be accomodated?

Reply:
It could be added to the space so it will depend on the space and avoid this issue